arrow-libs/fx/arrow-fx-coroutines/src/commonMain/kotlin/arrow/fx/coroutines/Schedule.kt

```diff
@@ -273,7 +274,7 @@ public sealed class Schedule<Input, Output> {
     zipDuration: (duration: Duration, otherDuration: Duration) -> Duration,
     zip: (Output, B) -> C
   ): Schedule<A, C> =
-    combineNanos(other, zipContinue, { a, b -> zipDuration(a.nanoseconds, b.nanoseconds).inNanoseconds }, zip)
+    combineNanos(other, zipContinue, { a, b -> zipDuration(a.nanoseconds, b.nanoseconds).toDouble(NANOSECONDS) }, zip)
```

Comment:
```suggestion
    combineNanos(other, zipContinue, { a, b -> zipDuration(a.nanoseconds, b.nanoseconds).inWholeNanoseconds }, zip)
```

Reply:
Hey @nomisRev I've used `.toDouble(NANOSECONDS)` to have the same type - `Double`. `inWholeNanoseconds` is way better, but it returns `Long` (probably that's the best thing in this method).
`combineNanos` expects `Double` (`zipDuration: (duration: Double, otherDuration: Double) -> Double`), so  using `.inWholeNanoseconds` won't compile. Let me know if I'm missing something here, I'm still learning Kotlin.

Reply:
Hey @wojda,
Oh, my bad then we should leave it as you made the fixes. I will resolve my comments as solved ;)
When we use `Duration` instead of `Double` internally for 2.0 we can improve this 👍

Reply:
Also, I took a look at `combineNanos`, it would be great to use `Duration` instead of `Double`. However, the method is public and just changing the type would be an API breaking change, not backward compatible.
I'm keen to take a look at Schedule.kt and replace `Double` with `Duration`. I would just need some guidance about ways of changing public API in Kotlin. Just break it ;) ? Create a new version of a method with Duration and deprecate Double? Something else?